### PR TITLE
feat: Adding tests for Safe Remote Purchase contract

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Adding tests for Safe Remote Purchase contract

**Related issue(s)**:
[[Solidity Support] Safe Remote Purchase examples#491](https://github.com/hashgraph/hedera-smart-contracts/issues/491)

### CODE REVIEW

File: test/solidity/safe-remote-purchase/safe-remote-purchase.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        const receiptAbort = await trxAbort.wait()
+
+        await ethers.provider.getBalance(seller.address)
+        const finalState = await contract.state()
+
+        expect(receiptAbort.events[0].event).to.equal('Aborted')
+        expect(finalState).to.equal(contractStates.Inactive)
+    })
+
+    describe('standart flow buyer -> seller tests: ', async function () {
```
Comment (ebadiere) [inline test/solidity/safe-remote-purchase/safe-remote-purchase.js:91]: Nit: Spelling, 'standard'

**Review by ebadiere — APPROVED** [review]
LG

File: artifacts/contracts/oz/access-control/AccessControlContract.sol/AccessControlContract.json [hunk 2]
```diff
@@ -0,0 +1,306 @@
+{
```
Comment (quiet-node) [inline artifacts/contracts/oz/access-control/AccessControlContract.sol/AccessControlContract.json:1]: Hmm have you rebased the main branch? Before the 0.7 release, I made sure that the "oz" folder has been moved to "openzeppelin" folder. These metadata files shouldn't be here

File: artifacts/contracts/oz/erc-4626/TokenVault.sol/TokenVault.json [hunk 3]
```diff
@@ -0,0 +1,926 @@
+{
```
Comment (quiet-node) [inline artifacts/contracts/oz/erc-4626/TokenVault.sol/TokenVault.json:1]: Hmm have you rebased the main branch? Before the 0.7 release, I made sure that the "oz" folder has been moved to "openzeppelin" folder. These metadata files shouldn't be here

File: artifacts/contracts/solidity/blind-auction/BlindAuction.sol/BlindAuction.json [hunk 4]
```diff
@@ -0,0 +1,298 @@
+{
```
Comment (quiet-node) [inline artifacts/contracts/solidity/blind-auction/BlindAuction.sol/BlindAuction.json:1]: Hmm have you rebased the main branch? Before the 0.7 release, I made sure that the "oz" folder has been moved to "openzeppelin" folder. These metadata files shouldn't be here

File: test/solidity/safe-remote-purchase/safe-remote-purchase.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    Release: 2,
+    Inactive: 3
+}
+const tinybarToHbarCoef = 100_000_000
+const EVEN_NUMBER_PRICE = 1 * tinybarToHbarCoef
+async function setupContract() {
+    const signers = await ethers.getSigners()
+    const seller = signers[0]
+    const buyer = signers[1]
+    const factoryPurchase = await ethers.getContractFactory('Purchase')
```
Comment (quiet-node) [inline test/solidity/safe-remote-purchase/safe-remote-purchase.js:36]: Please add the "Purchase" keyword to the `constants` file under [Contract object](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/constants.js#L75) and use the constant variable?

**Review by quiet-node — CHANGES_REQUESTED** [review]
Content LG just some Qs around

File: artifacts/contracts/solidity/blind-auction/BlindAuction.sol/BlindAuction.json (hunk 4 above)
Comment (stefan-stefanooov) [reply artifacts/contracts/solidity/blind-auction/BlindAuction.sol/BlindAuction.json:1]: Just did. Thank you.

File: artifacts/contracts/oz/access-control/AccessControlContract.sol/AccessControlContract.json (hunk 2 above)
Comment (quiet-node) [reply artifacts/contracts/oz/access-control/AccessControlContract.sol/AccessControlContract.json:1]: Hmm these metadata files are still here. I mean if there's only 1 new/modified contract should result into 1 new metadata file.

**Review by quiet-node — APPROVED** [review]
LGTM

**Pull request merged**